pkg/kube/multicluster/secretcontroller.go

```diff
@@ -130,9 +152,11 @@ func (c *Controller) AddHandler(h ClusterHandler) {
 func (c *Controller) Run(stopCh <-chan struct{}) error {
 	// run handlers for the local cluster; do not store this *Cluster in the ClusterStore or give it a SyncTimeout
 	// this is done outside the goroutine, we should block other Run/startFuncs until this is registered
-	localCluster := &Cluster{Client: c.localClusterClient, ID: c.localClusterID}
-	if err := c.handleAdd(localCluster, stopCh); err != nil {
-		return fmt.Errorf("failed initializing local cluster %s: %v", c.localClusterID, err)
+	if !features.LocalClusterSecretWatcher {
+		localCluster := &Cluster{Client: c.localClusterClient, ID: c.localClusterID}
```

Comment:
have you changed the localClusterClient meaning, if not why it has something to do with `features.LocalClusterSecretWatcher`